earth2studio/models/batch.py

```diff
@@ -164,15 +164,28 @@ def _batch_wrap(self, func: Callable) -> Callable:
         # TODO: Better typing for model object
         @functools.wraps(func)
         def _wrapper(
-            model: Any, x: torch.Tensor, coords: CoordSystem
+            model: Any,
+            x: torch.Tensor,
+            coords: CoordSystem,
+            conditioning: torch.Tensor | None = None,
+            conditioning_coords: CoordSystem | None = None,
         ) -> tuple[torch.Tensor, CoordSystem]:
 
             x, flatten_coords, batched_coords, batched_shape = self._compress_batch(
                 model, x, coords
             )
 
             # Model forward
-            out, out_coords = func(model, x, flatten_coords)
+            if conditioning is not None:
+                out, out_coords = func(
+                    model,
+                    x,
+                    flatten_coords,
+                    conditioning=conditioning,
+                    conditioning_coords=conditioning_coords,
+                )
+            else:
+                out, out_coords = func(model, x, flatten_coords)
```

Comment:
@NickGeneva if we force users to use a `conditioning_data_source` in the model then we can avoid making changes here.

Reply:
I agree thats better to force/expect the use to use a certain data source of fetch method.

imo its better to have that an explicit requirement in the model and call it out in the docs than change this behavior. Low key the batching behavior is already not so clear to users so I dont think its a good idea to further complicate its behavior for one model.

Reply:
Ok, so we want to remove the option entirely to pass the conditioning data at runtime to `StormCast` call? Upshot of doing that is it simplifies things and makes the call signature similar to other models (removing the need for the above changes); downside is probably performance, as then each `forward` will do it's own `fetch_data`

Reply:
No, `fetch_data` would be called from within each `__call__` when iterating over the generator. The model needs temporally-aligned conditioning data for each time-step

Reply:
Oh, I suppose I was thinking that the model would require all conditioning data to be fetched ahead of time as the input and then do the interpolation process at the beginning for all of it.

Yeah you could not run the generator forever like say FCN, but then you keep any downloading of data out of the model and all interpolation can occur at the beginning or even on the fly. But seems I'm miss understanding how this is working.

Reply:
But now that I think about it, you would have to pull the data remotely anyway regardless. You could also pre pull all data and then put in a local data source for the model to use. So maybe not that big of a deal

Reply:
K, pushed an update that removes these changes to the batching and requires `conditioning_data_source` to be set in `StormCast` before running the model